zerver/lib/utils.py

```diff
@@ -204,3 +206,27 @@ def check_subdomain(realm_subdomain, user_subdomain):
         if realm_subdomain != user_subdomain:
             return False
     return True
+
+def validate_domain(domain):
+        # type: (Text) -> Tuple[bool, str]
+        error_strings = {
+            'empty': _("Domain can't be empty."),
+            'too short': _("Domain needs to have length 3 or greater."),
+            'missing dot': _("Domain must have atleast one dot(.)."),
+            'extremal dash': _("Subdomain cannot start or end with a '-'."),
+            'extremal dot': _("Domain cannot start or end with a dot(.)."),
+            'bad character': _("Domain can only have lowercase letters, numbers, and '-'s.")}
```

Comment:
I would say letters instead of lowercase letters, and then always pass `domain.strip().lower()` into here and the database.